src/commands/run.rs

```diff
@@ -981,7 +977,16 @@ impl RunCommand {
             }
             #[cfg(feature = "wasi-http")]
             {
-                bail!("wasi-http support will be swapped over to component CLI support soon");
+                match linker {
+                    CliLinker::Core(linker) => {
+                        wasmtime_wasi_http::sync::add_to_linker(linker)?;
+                    }
+                    #[cfg(feature = "component-model")]
+                    CliLinker::Component(linker) => {
+                        wasmtime_wasi_http::proxy::sync::add_to_linker(linker)?;
```

Comment:
Using wasi-http with the component::Linker will fail because it conflicts with the stream methods in the wasmtime-wasi::preview2 implementation. So, we should instead `bail` in this case.

Reply:
that is actually not case because we are taking the implementation from the `wasmtime-wasi` crate when building the bindings: https://github.com/bytecodealliance/wasmtime/blob/75f3da7f029695787146cb89da5c8b3abc761738/crates/wasi-http/src/proxy.rs#L53-L54

we should be able to migrate the cli test to a component when the module code is deprecated.

Reply:
right, but without actually using HostInputStream and HostOutputStream, it will still crash at runtime.